Feature/ordem pagamento

lfdivino opened the pull request.

mileo committed:
[WIP] Ordem de pagamento da folha

hendrixcosta committed:
[ADD] Adicionar pagamento apartir do holerite

lfdivino committed:
[ADD] Ligacao entre holerite e linha de pagamento

Criado a ligacao entre um holerite para as 'n' linhas de uma ordem de
pagamento.

lfdivino committed:
[ADD] Aba das ordens de pagamento no holerite

Colocado o modulo l10n_br_hr_payment_order como dependencia do modulo.

lfdivino committed:
[ADD] Automatizar criacao de ordem de pagamento

Criado opcao no menu de configuracoes de contabilidade a opcao para
gerar automaticamente as ordens de pagamento dos holerites da folha de
pagamento.

lfdivino committed:
[ADD] Marcar rubrica a ser paga do holerite

Campo para verificar se a rubrica ira criar ordem de pagamento.

lfdivino committed:
[ADD] Wizard para busca das linhas dos holerites

Criado botao e o wizard para buscar as linhas pagaveis do(s) holerite(s)
para que seja criado ordens de pagamento.

hendrixcosta committed:
[ADD] Add partner nas payslip line

hendrixcosta committed:
[ADD] Calculo de rubricas especificas iguais

hendrixcosta committed:
[ADD] Partner na rubrica especifica para criar pagamentos em seu nome

hendrixcosta committed:
[FIX] Flake8

hendrixcosta committed:
[FIX] Flake8 Repositorio

hendrixcosta committed:
[ADD] Dependencia

hendrixcosta committed:
[FIX] Lançamento de ordem de pagamento para a Pensao calculada no holerite

hendrixcosta committed:
[ADD] regra de acesso

hendrixcosta committed:
[FIX] View de ordens de pagamento da folha

lfdivino committed:
[FIX] Xml das views da ordem de pagamento

Correcoes no xml das views de ordem de pagamento e no wizard para
carregar as linhas a serem pagas dos holerites no periodo selecionado.

lfdivino committed:
[ADD] Campos para a busca dos holerites no wizard

lfdivino committed:
[ADD] Funcoes para buscar linhas dos holerites

Funcoes para buscar as linhas a serem pagas nos holerites do periodo
selecionado no wizard.

lfdivino committed:
[ADD] Funcao para criar as ordens de pagamento

Criada funcoes para criar as linhas da ordem de pagamento a partir das
linhas selecionadas pelo wizard.

lfdivino committed:
[ADD] Tree view das linhas de holerite

Criado tree view para exibir as principais informacoes das linhas nos
holerites selecionados no determinado periodo pelo wizard, para melhor
visualizacao do usuario.

lfdivino committed:
fixup! [ADD] Funcao para criar as ordens de pagamento

Commit:
[ADD] states payment_line

Commit:
[ADD] states payment_line

Commit:
[FIX] variaveis de controle do workflow

Commit:
[FIX] variaveis de controle do workflow

Commit:
[ADD] workflow payment_line

Commit:
[ADD] workflow payment_line

Commit:
[FIX] workflow holerite

Commit:
[FIX] workflow holerite

Commit:
[ADD-WIP] status payment_line na visão

Commit:
[ADD-WIP] status payment_line na visão

Commit:
[FIX] referencias no workflow do holerite

Commit:
[FIX] referencias no workflow do holerite

Commit:
[FIX] troca de state para status

Commit:
[FIX] troca de state para status

Commit:
[FIX] troca de state para status

Commit:
[FIX] troca de state para status

Commit:
[ADD] módulo cnab_payslip

Commit:
[FIX] nome campo

Commit:
[FIX] nome campo

Commit:
[FIX-WIP] botão 'done' para holerite

Commit:
[FIX-WIP] botão 'done' para holerite

Commit:
[REM] status payment_line

Commit:
[REM] status payment_line

Commit:
[REM] status payment_line

Commit:
[REM] status payment_line

lfdivino committed:
[FIX] Workflow do holerite da folha de pagamento

Holerite agora para no state 'verify' e foi adicionado o botao para
avancar para o state 'done'

lfdivino committed:
[FIX] Workflow da folha de pagamento

Corrigido o workflow para alterar o state do holerite para concluido
quando todas as payment.lines associadas fiquem no 'state2' paid.

Commit:
[FIX] permissão de criação wizard holerite do payment_order

Commit:
[FIX] state no search de done para verify

gabrielcardoso21 committed:
[ADD] Campo conta_bancaria_id no contrato

Hendrixcosta commented on l10n_br_hr_payment_order/views/hr_payslip.xml:
```diff
@@ -15,21 +15,20 @@
             <button name="refund_sheet" position="after">
                 <button name="create_payment_order" type="object"
                         string="Add to Payment Order" />
+                <button name="action_done" type="object"
```
Esses buttons não estão duplicados?

Hendrixcosta commented on l10n_br_hr_payment_order/models/payment_order.py:
```diff
@@ -5,11 +5,6 @@
 from openerp import fields, models
 
 
-class PaymentOrder(models.Model):
```
Arquivo tem que ter nome do modelo implementado na classe.

lfdivino committed:
[FIX] Cancel workflow da payment.order

lfdivino committed:
[FIX] Tornar invisivel botao voltar para rascunho

lfdivino committed:
fixup! [FIX] Cancel workflow da payment.order

lfdivino committed:
[FIX] Lote de holerites

Agora os holerites gerados pelo lote ja sao confirmados e ficam no state
de 'Em espera'

lfdivino committed:
[REF] Codigo para fechamento do holerite

Transferido o codigo do workflow para fechamento do holerite quando a(s)
payment.line estiverem pagas do modulo do cnab_payslip para o
l10n_br_hr_payment_order. Exluido o modulo cnab_payslip

hendrixcosta committed:
[ADD] Setar regras de liquido de salario e de pensao como regras que devem ser pagas pelas ordens de pagamento

hendrixcosta committed:
[FIX] Erro de singleton ao atualizar modulo

hendrixcosta committed:
[IMP] Dependencias externas

hendrixcosta committed:
[FIX] Instalação cnab travis

mileo commented on l10n_br_hr_payment_order/models/hr_payslip.py:
```diff
@@ -0,0 +1,126 @@
+        payment_line_model = self.env['payment.line']
+        vals = {
+            'order_id': payment_order.id,
+            # 'partner_bank_id': self.partner_bank_id.id,
```
Isto esta correto ?

hendrixcosta replied:
Feito Ajuste.

mileo commented on l10n_br_hr_payment_order/models/hr_payslip.py:
```diff
@@ -0,0 +1,126 @@
+            # Buscar ordens de pagamento do mesmo tipo
+            payorders = payment_order_model.search([
+                ('mode', '=', holerite.payment_mode_id.id),
+                ('state', '=', 'draft')]
```
Estado incorreto

hendrixcosta replied:
Busca das payment_order que estão em aberto. Estado é draft mesmo

mileo commented on l10n_br_hr_payment_order/views/hr_contract.xml:
```diff
@@ -0,0 +1,14 @@
+            </field>
+        </record>
+    </data>
+</openerp>
```
New Line

hendrixcosta committed:
Update hr_contract.xml

hendrixcosta committed:
[FIX] String de campos

hendrixcosta committed:
[FIX] Ajuste para pegar informaçoes bancarias

hendrixcosta committed:
[REM] Arquivo obsoleto

Commit:
[ADD] Confirmar todos os Holerites de um Lote no botão Fechar Lote

hendrixcosta commented:
@wagnerpereirasp por favor estabilize o travis

Commit:
fixup! [ADD] Confirmar todos os Holerites de um Lote no botão Fechar Lote
